Observer/PreventCustomerEdit.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace Bolt\Boltpay\Observer;
+
+use Bolt\Boltpay\Helper\FeatureSwitch\Decider;
+use Magento\Framework\App\Action\Action;
+use Magento\Framework\App\ActionFlag;
+use Magento\Framework\App\Response\RedirectInterface;
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Event\ObserverInterface;
+use Magento\Framework\HTTP\PhpEnvironment\Request;
+use Magento\Framework\Message\ManagerInterface;
+use Magento\Framework\UrlInterface;
+
+class PreventCustomerEdit implements ObserverInterface
+{
+    /**
+     * @var ActionFlag
+     */
+    private $actionFlag;
+
+    /**
+     * @var RedirectInterface
+     */
+    private $redirect;
+
+    /**
+     * @var UrlInterface
+     */
+    private $urlManager;
+
+    /**
+     * @var ManagerInterface
+     */
+    private $messageManager;
+
+    /**
+     * @var Decider
+     */
+    private $featureSwitches;
+
+    /**
+     * @param ActionFlag $actionFlag
+     * @param RedirectInterface $redirect
+     * @param UrlInterface $urlManager
+     * @param ManagerInterface $messageManager
+     * @param Decider $featureSwitches
+     */
+    public function __construct(
+        ActionFlag        $actionFlag,
+        RedirectInterface $redirect,
+        UrlInterface      $urlManager,
+        ManagerInterface  $messageManager,
+        Decider           $featureSwitches
+    ) {
+        $this->actionFlag = $actionFlag;
+        $this->redirect = $redirect;
+        $this->urlManager = $urlManager;
+        $this->messageManager = $messageManager;
+        $this->featureSwitches = $featureSwitches;
+    }
+
+    /**
+     * @param Observer $observer
+     *
+     * @return void
+     */
+    public function execute(Observer $observer)
+    {
+        if (!$this->featureSwitches->isPreventSSOCustomersFromEditingAccountInformation()) {

```

Comment:
@danac-gs @vitaliyreznikov Should we add one more condition **|| isBoltSSOEnabled()** ?

Reply:
Done